pkg/scorecard/statement.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2025 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Subject {
+// 	 Repo           jsonRepoV2          `json:"repo"`
+// }
+
+package scorecard
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	intoto "github.com/in-toto/attestation/go/v1"
+
+	docs "github.com/ossf/scorecard/v5/docs/checks"
+	sce "github.com/ossf/scorecard/v5/errors"
+)
+
+const (
+	PredicateType = "https://scorecard.dev/result/v1"
+)
+
+type Statement struct {
+	intoto.Statement
+	Predicate Predicate `json:"predicate"`
+}
+
+// Predicate overrides JSONScorecardResultV2 with a nullable Repo field.
+type Predicate struct {
+	Repo *jsonRepoV2 `json:"repo,omitempty"`
+	JSONScorecardResultV2
+}
+
+// AsStatement converts the results as an int-toto statement.
+func (r *Result) AsStatement(writer io.Writer, checkDocs docs.Doc, opt *AsJSON2ResultOption) error {
+	// Attestatio subject
```

Comment:
do we envision any configuration we'd want to do independent of the JSON where we should we define an AsStatementOption (or `AsIntotoOption` depending on other comment)?

(also: typo in `Attestation`)

Reply:
Good idea. Right now, there is nothing to change, but I think we'll iterate the predicate or add other types, so specifying the predicate type will probably be the first option. I've wrapped the JSON2 options in a new AsStatement options set (which is empty for now).